src/main/java/com/loofah/graph/api/services/SkillService.java

```diff
@@ -2,55 +2,34 @@
 
 import com.loofah.graph.api.models.database.Skill;
 import com.loofah.graph.api.models.filters.SkillFilter;
+import com.loofah.graph.api.providers.MongoQueryProvider;
 import com.loofah.graph.api.repositories.SkillRepository;
-import org.jetbrains.annotations.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.mongodb.core.query.Criteria;
 import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.stereotype.Component;
 
 import java.util.List;
-import java.util.Optional;
 
 @Component
 public class SkillService {
 
     private final SkillRepository skillRepository;
+    private final MongoQueryProvider mongoQueryProvider;
 
     @Autowired
-    public SkillService(SkillRepository skillRepository) {
+    public SkillService(final SkillRepository skillRepository,
+                        final MongoQueryProvider mongoQueryProvider) {
         this.skillRepository = skillRepository;
+        this.mongoQueryProvider = mongoQueryProvider;
     }
 
     public Skill getById(String id) {
         return skillRepository.findById(id).get();
     }
 
     public List<Skill> getWithFilter(SkillFilter skillFilter) {
-        Query query = buildQuery(skillFilter);
+        Query query = mongoQueryProvider.buildMongoQuery(skillFilter);
```

Comment:
Nitpick: this service class still has a dependency on our database implementation (i.e. it's depending on `Query` and`MongoQueryProvider` classes, and on a `SkillRepository` interface which extends a mongo interface). This means that if we ever do want to change the database implementation, this service will have to change too.

Well beyond the scope of this ticket to fix this, but something for us to bear in mind.

Reply:
As discussed, great shout and will bring through another ticket immediately after this one to improve the design a bit here